### DISCUSSION TITLE
How to use  uploading files in custom operation?

### DISCUSSION THREAD

**Discussion by rezahmady** [Laravel-Backpack/community-forum#1091] [General]
When creating a New Operation with a Form how to uploads files?

**Comment by jcastroa87** [comment]
Hello @rezahmady 

What field are you trying to use? How your custom operation works?

Cheers.

**Comment by tabacitu** [comment]
Sorry for the late reply @rezahmady .

We don't have a guide on how to use uploaders in custom operations - but if more people ask for it, we could create one. Right now the only way to do that is to look at the code of the Create & Update operations and do the same thing. 

If you take a look at [the Create operation](https://github.com/Laravel-Backpack/CRUD/blob/main/src/app/Http/Controllers/Operations/CreateOperation.php) for example, you can see we're calling methods on the CrudPanel object:

```php
    public function store()
    {
        $this->crud->hasAccessOrFail('create');

        // execute the FormRequest authorization and validation, if one is required
        $request = $this->crud->validateRequest();

        // register any Model Events defined on fields
        $this->crud->registerFieldEvents();

        // insert item in the db
        $item = $this->crud->create($this->crud->getStrippedSaveRequest($request));
        $this->data['entry'] = $this->crud->entry = $item;

        // show a success message
        \Alert::success(trans('backpack::crud.insert_success'))->flash();

        // save the redirect choice for next time
        $this->crud->setSaveAction();

        return $this->crud->performSaveAction($item->getKey());
    }
```

Go deeper into the CrudPanel `create()` method if you want to see HOW that is done. But for your purpose, I think just using the same methods inside the CreateOperation and UpdateOperation will help - as long as you use the same methods and give the same fields, the Uploaders should work 

Hope it helps.
Cheers!
